designs/service_function_chaining/SFC_design.md

```diff
@@ -0,0 +1,127 @@
+- Flow-Classification : A flow classifier identifies a flow. A port chain can contain multiple flow classifiers. Omitting the flow classifier effectively prevents steering of traffic through the port chain.  
+
+### SFC Configuration
+- SFC is neutron extension in Openstack compute node
+- OVS will be used as backend driver
+- Current configuration on Ocata red hat Openstack platform.
+
+SFC work flow in Openstack Environment.
+![](./images/sfc-dataflow.png)
+
+
+## Design Changes  
+SFC functionality in OSC will adapt to existing SDN controller SDK APIs.SFC configuration work flow, changes to SDK APIs and interfaces to support SFC plugin are described below in detail. Other supported SDN controller plugins should adapt to all the below mention changes.
+
+![](./images/sw-modules.png)
+### REST API   
+Currently not applicable.
+ 
+### OSC SDKs  
+
+#### SDN Controller SDK  
+Following section describes in detail changes in SDN controller SDK APIs and interfaces to support SFC plugin.
+
+
+##### Inspection port
+OSC core will create inspection port during creation of new distribution specification (DS) under a distribution Appliance.  
+
+SDN controller SDK APIs `registerinspectionport` and `removeinspectionport` will be called to create and remove the inspection port with `InspectionPortElement` interface as the argument.  
+
+SFC plugin should adapt the same SDK controller APIs to create and delete inspection port hook entity in OSC. 
+
+Following changes are proposed with respect to Inspection port actions.
+
+`InspectionPortElement` interface will have an additional field `redirection-target-id`. For more understanding of `redirection-target-id` follow inspection port work flow described below.
+
+This `redirection-target-id` will be used as Inspection port for SFC plugin.
+
+```
+Current Inspection port hook APIs
+string registerInspectionPort(InspectionPortElement inspectionPort) throws NetworkPortNotFoundException, Exception;
+```
+```
+Proposed Inspection port hook APIs
+InspectionPortElement registerInspectionPort(InspectionPortElement inspectionPort) throws NetworkPortNotFoundException, Exception;
+```
+Current InspectionPortElement inspectionPort interface data structure
+```
+public interface InspectionPortElement {
+
+    NetworkElement getIngressPort();
+
+    NetworkElement getEgressPort();
+}
+```
+
+Proposed changes to InspectionPortElement inspectionPort interface
+```
+public interface InspectionPortElement {
+
+    NetworkElement getIngressPort();
+
+    NetworkElement getEgressPort();
+
+	
+	String redirection-target-id();			// Parent ID will be set with port-pair-group id, return from openstack 
+    											afte the port-pair is either added or updated to port-pair-group.(PPG)
+}
+```
+
+Inspection port work flow is described in the following diagram. 
+
+![](./images/reg-inspection-port.png)

```

Comment:
in the diagram, ds-inspection port seems confusing. shouldnt this be dai-inspection port?